integration-tests/src/integrationTest/java/pl/allegro/tech/hermes/integrationtests/AsyncPublisherAutoRetryTest.java

```diff
@@ -0,0 +1,74 @@
+package pl.allegro.tech.hermes.integrationtests;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.web.reactive.function.client.WebClient;
+import pl.allegro.tech.hermes.client.HermesClient;
+import pl.allegro.tech.hermes.client.HermesClientBuilder;
+import pl.allegro.tech.hermes.client.HermesSender;
+import pl.allegro.tech.hermes.client.webclient.WebClientHermesSender;
+import pl.allegro.tech.hermes.test.helper.util.Ports;
+
+import java.net.URI;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+
+public class AsyncPublisherAutoRetryTest {
+    private static final String TOPICS_ROOT = "/topics/";
+    private static final String EXISTING_TOPIC = "existing";
+    private static final String EXISTING_TOPIC_URL = TOPICS_ROOT + EXISTING_TOPIC;
+    private static final String NOT_EXISTING_TOPIC = "not-existing";
+    private static final String NOT_EXISTING_TOPIC_URL = TOPICS_ROOT + NOT_EXISTING_TOPIC;
+    private static final int frontendPort = Ports.nextAvailable();
```

Comment:
I wouldn't use `Ports.nextAvailable()`. Take a look at how it is implemented. It opens and closes a socket, potentially leaving it in the `TIME_WAIT` state. This, in turn, sometimes leads to "Address already in use" errors. Instead, I'd use ephemeral ports - just pass 0 as a port.

Reply:
good catch, applied the suggestion in another test